modules/currency-l0/src/main/scala/io/constellationnetwork/currency/l0/StateChannel.scala

```diff
@@ -24,17 +25,24 @@ object StateChannel {
   def run[F[_]: Async: HasherSelector](
     services: Services[F, Run],
     storages: Storages[F],
-    programs: Programs[F]
+    programs: Programs[F],
+    dataApplicationService: Option[BaseDataApplicationL0Service[F]]
   )(implicit S: Supervisor[F]): Stream[F, Unit] = {
     val logger = Slf4jLogger.getLogger[F]
 
-    def globalL0SnapshotProcessing: Stream[F, Unit] =
+    def globalL0SnapshotProcessing: Stream[F, Unit] = {
+      def triggerOnGlobalSnapshotPullHook(snapshot: Hashed[GlobalIncrementalSnapshot], context: GlobalSnapshotInfo) =
+        dataApplicationService match {
+          case Some(service) => service.onGlobalSnapshotPull(snapshot, context)
```

Comment:
I'd put there an error handler for unexpected errors from `onGlobalSnapshotPull`